[Writing Tools] Support compiling ENABLE_WRITING_TOOLS on the public SDK (Part 4)

rr-codes opened the pull request:
#### ce1eafc8bc2caf6669a00722b1ac06e7edb3e74d
<pre>
[Writing Tools] Support compiling ENABLE_WRITING_TOOLS on the public SDK (Part 4)
<a href="https://bugs.webkit.org/show_bug.cgi?id=291811">https://bugs.webkit.org/show_bug.cgi?id=291811</a>
<a href="https://rdar.apple.com/149644885">rdar://149644885</a>

Reviewed by Aditya Keerthi.

More iOS-specific work for ENABLE_WRITING_TOOLS on the public SDK.

Since some WebKit Swift files use Swift UIKit SPI, a dummy `.swiftinterface` file and module must be
created so that the symbols can be found when compiling with the public SDK, similar to the existing
`QuickLook_SPI` module.

* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsUISPI.h:
* Tools/TestWebKitAPI/Tests/TestWebKitAPI/mac/AppKitSPI.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:

Drive-by fix: add some missing forward declarations and imports.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

Add the `ios` SPI directory to the Swift include paths, similar to the existing `visionos` directory included.

* Source/WebKit/Platform/spi/ios/UIKit_SPI.swiftinterface: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Make a manually-created swiftinterface file similar to the real UIKit and UIKitCore swiftinterfaces,
only including the symbols used in WebKit/WebKitSwift, and using a unique module name `UIKit_SPI`.

* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift:
* Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:

When compiling with the public SDK, use the `UIKit_SPI` module so that the symbols can be accessed.
Also remove an unused import.

Canonical link: <a href="https://commits.webkit.org/293923@main">https://commits.webkit.org/293923@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/482e2062f8c2a3086f00a2f1d75a7af247dfc626

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100260 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19909 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/10207 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105390 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50843 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/102301 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20214 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28383 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76333 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33391 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103267 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15464 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90560 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56690 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/15284 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8559 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/50211 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/85194 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8641 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107749 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27373 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20068 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85285 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27736 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86767 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84824 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21572 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29508 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7243 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/21259 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27310 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32543 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27121 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30437 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28680 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/Platform/spi/ios/UIKit_SPI.swiftinterface:
```diff
@@ -0,0 +1,134 @@
+@objc @_hasMissingDesignatedInitializers @available(iOS 18.0, visionOS 2.0, *)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@_Concurrency.MainActor public class UIDirectionalLightEffectView : UIKit.UIView {
```
Is `UIDirectionalLightEffectView` necessary?

rr-codes replied:
It is, for the default parameter value for one of the initializers above, which itself is needed so that the disfavoredOverload attribute works as expected

rr-codes committed:
[Writing Tools] Support compiling ENABLE_WRITING_TOOLS on the public SDK (Part 4)
https://bugs.webkit.org/show_bug.cgi?id=291811
rdar://149644885

Reviewed by Aditya Keerthi.

More iOS-specific work for ENABLE_WRITING_TOOLS on the public SDK.

Since some WebKit Swift files use Swift UIKit SPI, a dummy `.swiftinterface` file and module must be
created so that the symbols can be found when compiling with the public SDK, similar to the existing
`QuickLook_SPI` module.

* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/WritingToolsUISPI.h:
* Tools/TestWebKitAPI/Tests/TestWebKitAPI/mac/AppKitSPI.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:

Drive-by fix: add some missing forward declarations and imports.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

Add the `ios` SPI directory to the Swift include paths, similar to the existing `visionos` directory included.

* Source/WebKit/Platform/spi/ios/UIKit_SPI.swiftinterface: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Make a manually-created swiftinterface file similar to the real UIKit and UIKitCore swiftinterfaces,
only including the symbols used in WebKit/WebKitSwift, and using a unique module name `UIKit_SPI`.

* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift:
* Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:

When compiling with the public SDK, use the `UIKit_SPI` module so that the symbols can be accessed.
Also remove an unused import.

Canonical link: https://commits.webkit.org/293923@main
